eth/stagedsync/stage_headers.go

```diff
@@ -95,16 +95,6 @@ func SpawnStageHeaders(
 	initialCycle bool,
 	test bool, // Set to true in tests, allows the stage to fail rather than wait indefinitely
 ) error {
-	useExternalTx := tx != nil
```

Comment:
Why did you remove this code?

Reply:
no transactions in HeadersPOS will be external

Reply:
we may need a write transaction for proposing

Reply:
Whether `tx` is nil or not depends on `canRunCycleInOneTransaction` in `StageLoopStep`, doesn't it? So `SpawnStageHeaders` may be called with `tx == nil`, right? Then `rawdb.Transitioned(tx, ...)`at line 106 will fail.

Reply:
Oh, okay fair,might have removed it by mistake. fixed